### PR TITLE
Merge main into preview part 2

### PR DESCRIPTION
**Pull request by timtay-microsoft**

### CODE REVIEW

File: e2e/test/iothub/FileUploadE2ETests.cs [hunk 1]
```diff
@@ -55,7 +55,8 @@ public async Task FileUpload_SmallFile_GranularSteps_ValidCorrelationId()
             await UploadFileGranularAsync(fileStreamSource, filename, fileUploadTransportSettings, isCorrelationIdValid: true).ConfigureAwait(false);
         }
 
-        [LoggedTestMethod, Timeout(LongRunningTestTimeoutMilliseconds)]
+        [LoggedTestMethod]
+        [Timeout(TestTimeoutMilliseconds)]
```
Comment (timtay-microsoft) [inline e2e/test/iothub/FileUploadE2ETests.cs:59]: to revert

File: e2e/test/iothub/messaging/FaultInjectionPoolAmqpTests.MessageSendFaultInjectionPoolAmqpTests.cs [hunk 2]
```diff
@@ -202,6 +202,7 @@ public async Task Message_ThrottledConnectionRecovery_MultipleConnections_AmqpWs
 
         [LoggedTestMethod]
         [Timeout(LongRunningTestTimeoutMilliseconds)]
+        [ExpectedException(typeof(UnauthorizedException))]
```
Comment (timtay-microsoft) [inline e2e/test/iothub/messaging/FaultInjectionPoolAmqpTests.MessageSendFaultInjectionPoolAmqpTests.cs:205]: revert

File: e2e/test/iothub/messaging/FaultInjectionPoolAmqpTests.MessageSendFaultInjectionPoolAmqpTests.cs [hunk 3]
```diff
@@ -225,7 +226,8 @@ public async Task Message_AuthenticationNoRecovery_MultipleConnections_Amqp()
 
         [LoggedTestMethod]
         [Timeout(LongRunningTestTimeoutMilliseconds)]
-        public async Task Message_IoTHubSak_AuthenticationNoRecovery_MultipleConnections_AmqpWs()
+        [ExpectedException(typeof(UnauthorizedException))]
```
Comment (timtay-microsoft) [inline e2e/test/iothub/messaging/FaultInjectionPoolAmqpTests.MessageSendFaultInjectionPoolAmqpTests.cs:229]: revert

File: e2e/test/iothub/service/IoTHubServiceProxyE2ETests.cs [hunk 4]
```diff
@@ -64,7 +64,6 @@ public async Task ServiceClientDevices_AddAndRemoveDevice_WithProxy()
 
         [LoggedTestMethod]
         [Timeout(TestTimeoutMilliseconds)]
-        [Ignore]
```
Comment (timtay-microsoft) [inline e2e/test/iothub/service/IoTHubServiceProxyE2ETests.cs:67]: revert

**Review by brycewang-microsoft — APPROVED** [review]

**Pull request merged**